iterator/iterator.c

```diff
@@ -2811,6 +2812,32 @@ processQueryTargets(struct module_qstate* qstate, struct iter_qstate* iq,
 			iq->dnssec_expected?"expected": "not expected",
 			iq->dnssec_lame_query?" but lame_query anyway": "");
 	}
+
+	struct sockaddr_storage real_addr = target->addr;
+	socklen_t real_addrlen = target->addrlen;
+
+	if (ie->use_nat64 && real_addr.ss_family == AF_INET) {
+		struct sockaddr_in *sin = (struct sockaddr_in *)&target->addr;
+		struct sockaddr_in6 *sin6;
+
+		real_addr = ie->nat64_prefix_addr;
+		real_addrlen = ie->nat64_prefix_addrlen;
+
+		/* - NAT64 prefix length really doesn't matter at all,
+		 *   it's just a bunch of zero padding
+		 * - config validation requires the NAT64 prefix to be /96
```

Comment:
Whoops, I completely forgot that RFC6052 actually specifies a different encoding for the prefix lengths. Fixing that in a minute.